### ISSUE TITLE
Combined loss does not match

### ISSUE THREAD

**Issue by mangleddata** [ludwig-ai/ludwig#1190]
Not sure if this is expected or not, but for single output feature I see the reported loss value don't match with combined value. It looks like below
```
[1,0]<stdout>:╒════════════════════════════╤════════╤════════════╕
[1,0]<stdout>:│ Predictor                  │   loss │   accuracy │
[1,0]<stdout>:╞════════════════════════════╪════════╪════════════╡
[1,0]<stdout>:│ train                      │ 0.6029 │     0.6786 │
[1,0]<stdout>:├────────────────────────────┼────────┼────────────┤
[1,0]<stdout>:│ vali                       │ 0.7369 │     0.5585 │
[1,0]<stdout>:╘════════════════════════════╧════════╧════════════╛
[1,0]<stdout>:╒════════════╤════════╕
[1,0]<stdout>:│ combined   │   loss │
[1,0]<stdout>:╞════════════╪════════╡
[1,0]<stdout>:│ train      │ 0.6241 │
[1,0]<stdout>:├────────────┼────────┤
[1,0]<stdout>:│ vali       │ 0.7066 │
[1,0]<stdout>:╘════════════╧════════╛
[1,0]<stdout>:Validation loss on combined improved, model saved
[1,0]<stdout>:
[1,0]<stdout>:
```

In some cases, the combined loss corresponding to validation set is significantly different than other metric. Any idea ?

**Comment by w4nderlust** [comment]
@mangleddata  did you get this using Ludwig 0.3.3? This PR should have fixed this already: https://github.com/ludwig-ai/ludwig/pull/1103

Anyway, in some cases the combined loss may still not match. For instance that is true if you have regularization, as the combined loss is the weighted sum of the losses of each output plus regularization.

Closing for now, let me know if the problem persists on master.
